### ISSUE TITLE
sort-imports not behaving as expected

### ISSUE THREAD

**Issue by aminnairi** [eslint/eslint#13222]
**Tell us about your environment**

* **ESLint Version:** 6.8.0
* **Node Version:** 13.12.0
* **npm Version:** 6.14.4

**What parser (default, Babel-ESLint, etc.) are you using?** `@typescript-eslint/parser`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "env": {
        "es6": true,
        "node": true
    },
    "extends": [
        "@aminnairi",
        "plugin:@typescript-eslint/eslint-recommended"
    ],
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
        "no-new": "off",
        "no-unused-vars": "off",
        "no-ternary": "off"
    }
}
```

*Extract from aminnairi/eslint-config*

```json
"sort-imports": ["error", { "ignoreCase": false, "ignoreDeclarationSort": false, "ignoreMemberSort": false, "memberSyntaxSortOrder": ["none", "all", "single", "multiple"] }],
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```typescript
import Breadcrumbs from "@material-ui/core/Breadcrumbs";
import Button from "@material-ui/core/Button";
import Container from "@material-ui/core/Container";
import FormControl from "@material-ui/core/FormControl";
import GitHubIcon from "@material-ui/icons/GitHub";
import Grid from "@material-ui/core/Grid";
import InputLabel from "@material-ui/core/InputLabel";
import {Link} from "react-router-dom";
import MenuItem from "@material-ui/core/MenuItem";
import Select from "@material-ui/core/Select";
import SnackbarContent from "@material-ui/core/SnackbarContent";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";
import {makeStyles} from "@material-ui/core/styles";
import React, {ChangeEvent, MouseEvent, SyntheticEvent, useState} from "react";
import Snackbar, {SnackbarOrigin} from "@material-ui/core/Snackbar";
```

```bash
$ eslint --fix
```

**What did you expect to happen?**

The named import for `Link` should be sorted with the named imports, not with the default imports.

**What actually happened? Please include the actual, raw output from ESLint.**

The named import for `Link` is not sorted with its named imports brothers.

**Are you willing to submit a pull request to fix this bug?**

No for the moment since I've never dived into the source-code of ESLint yet.

**Comment by anikethsaha** [comment]
I think It is working as expected. 
Your order of sort is `["none", "all", "single", "multiple"]`, so that `{Link}` is single and uppercase as your config has `ignoreCase : false`, it will prioritize the uppercase over the lowercase so it is correct order. 

`import {makeStyles} from "@material-ui/core/styles";`
is not with `{Link}` because between them, all the uppercase so those will be sorted first and as all of them are `single` so as `{Link}`, they will be grouped.

**Comment by aminnairi** [comment]
Thanks for your answer @anikethsaha.

I see, do you know what set of rules I should use in order to get the intended behaviour, meaning having my `import {Link}` sorted with named imports first?

**Comment by anikethsaha** [comment]
I think currently there is no option to group them. 
for this specific snippet, you can try `ignoreCase: true` 

[demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHNvcnQtaW1wb3J0czogW1wiZXJyb3JcIix7IFwiaWdub3JlQ2FzZVwiOiB0cnVlLCBcImlnbm9yZURlY2xhcmF0aW9uU29ydFwiOiBmYWxzZSwgXCJpZ25vcmVNZW1iZXJTb3J0XCI6IGZhbHNlLCBcIm1lbWJlclN5bnRheFNvcnRPcmRlclwiOiBbXCJub25lXCIsIFwiYWxsXCIsIFwic2luZ2xlXCIsIFwibXVsdGlwbGVcIl0gfV0gKi9cbmltcG9ydCBCcmVhZGNydW1icyBmcm9tIFwiQG1hdGVyaWFsLXVpL2NvcmUvQnJlYWRjcnVtYnNcIjtcbmltcG9ydCBCdXR0b24gZnJvbSBcIkBtYXRlcmlhbC11aS9jb3JlL0J1dHRvblwiO1xuaW1wb3J0IENvbnRhaW5lciBmcm9tIFwiQG1hdGVyaWFsLXVpL2NvcmUvQ29udGFpbmVyXCI7XG5pbXBvcnQgRm9ybUNvbnRyb2wgZnJvbSBcIkBtYXRlcmlhbC11aS9jb3JlL0Zvcm1Db250cm9sXCI7XG5pbXBvcnQgR2l0SHViSWNvbiBmcm9tIFwiQG1hdGVyaWFsLXVpL2ljb25zL0dpdEh1YlwiO1xuaW1wb3J0IEdyaWQgZnJvbSBcIkBtYXRlcmlhbC11aS9jb3JlL0dyaWRcIjtcbmltcG9ydCBJbnB1dExhYmVsIGZyb20gXCJAbWF0ZXJpYWwtdWkvY29yZS9JbnB1dExhYmVsXCI7XG5pbXBvcnQge0xpbmt9IGZyb20gXCJyZWFjdC1yb3V0ZXItZG9tXCI7XG5pbXBvcnQge21ha2VTdHlsZXN9IGZyb20gXCJAbWF0ZXJpYWwtdWkvY29yZS9zdHlsZXNcIjtcbmltcG9ydCBNZW51SXRlbSBmcm9tIFwiQG1hdGVyaWFsLXVpL2NvcmUvTWVudUl0ZW1cIjtcbmltcG9ydCBTZWxlY3QgZnJvbSBcIkBtYXRlcmlhbC11aS9jb3JlL1NlbGVjdFwiO1xuaW1wb3J0IFNuYWNrYmFyQ29udGVudCBmcm9tIFwiQG1hdGVyaWFsLXVpL2NvcmUvU25hY2tiYXJDb250ZW50XCI7XG5pbXBvcnQgVGV4dEZpZWxkIGZyb20gXCJAbWF0ZXJpYWwtdWkvY29yZS9UZXh0RmllbGRcIjtcbmltcG9ydCBUeXBvZ3JhcGh5IGZyb20gXCJAbWF0ZXJpYWwtdWkvY29yZS9UeXBvZ3JhcGh5XCI7XG5pbXBvcnQgUmVhY3QsIHtDaGFuZ2VFdmVudCwgTW91c2VFdmVudCwgU3ludGhldGljRXZlbnQsIHVzZVN0YXRlfSBmcm9tIFwicmVhY3RcIjtcbmltcG9ydCBTbmFja2Jhciwge1NuYWNrYmFyT3JpZ2lufSBmcm9tIFwiQG1hdGVyaWFsLXVpL2NvcmUvU25hY2tiYXJcIjsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7InNvcnQtaW1wb3J0cyI6Mn0sImVudiI6e319fQ==)

maybe, it can be a feature request ( rule change behind an option ). but IMO, it will increase the complexity as named imports do have members so  for `memberSyntaxSortOrder` , having option like

`["none", "all", "named", "single", "multiple"]`

it will create inconsistency among `named` and `multiple`. 

Lets hear from others as I may miss something.

**Comment by mdjermanovic** [comment]
@aminnairi thanks for the issue!

This works as intended, because `"single"` and `"multiple"` are just counting imported names.

I'd personally support an enhancement to have `"default"` and `"named"` as an alternative to  `"single"` and `"multiple"`, like #12065. Please feel free to open a rule change request.

> I see, do you know what set of rules I should use in order to get the intended behaviour, meaning having my import {Link} sorted with named imports first?

Unfortunately, I don't think there is a way to achieve this at the moment, other than with `eslint-disable` comments (or e.g., blank lines when #12956 gets implemented).

**Comment by aminnairi** [comment]
Thanks for the time you took to answer my really a question more than an issue at this point.

I'll try to get my head into this project and when I'll feel confident enough I'll let you know if I can find a workaround.

I think it's safe now to say that this "issue" is closed now.

**Comment by anikethsaha** [comment]
> I'd personally support an enhancement to have `"default"` and `"named"` as an alternative to `"single"` and `"multiple"`, like #12065. Please feel free to open a rule change request.

I too think `named` and `default` are much better.

**Comment by jbnv** [comment]
I support this as well. I just spent several minutes and almost submitted a Stack overflow question because of this. This is what I expect and want:

```
import { always, cond, is, isNil, path, T } from 'ramda';
import { BLink } from 'bootstrap-vue';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import alignments from '~/data/alignments.json';
import events from '~/data/events.json';
```

This is what flags no errors, but is not what I wanted or asked for:

```
import { always, cond, is, isNil, path, T } from 'ramda';
import alignments from '~/data/alignments.json';
import { BLink } from 'bootstrap-vue';
import events from '~/data/events.json';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
```
It does not "work as expected." The documentation shows nothing about this case and clearly gives the impression that `import { named }` would be sorted before `import default`.

**Comment by mdjermanovic** [comment]
> It does not "work as expected." The documentation shows nothing about this case and clearly gives the impression that `import { named }` would be sorted before `import default`.

Indeed, we don't have any examples with a single named import. 

PR to improve the docs and clarify the actual behavior is welcome!
